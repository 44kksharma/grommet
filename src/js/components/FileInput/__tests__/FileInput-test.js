import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import { FileInput } from '..';

describe('FileInput', () => {
  test('basic', () => {
    const { container } = render(<FileInput name="file" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('multiple', () => {
    const { container } = render(<FileInput name="file" multiple />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('accept', () => {
    const { container } = render(<FileInput name="file" accept="image/*" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('disabled', () => {
    const { container } = render(<FileInput name="file" disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
