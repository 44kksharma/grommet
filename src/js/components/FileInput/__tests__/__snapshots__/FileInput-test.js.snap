// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput accept 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: dashed 4px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.c3 {
  position: relative;
  display: block;
  pointer-events: none;
}

.c6 {
  margin: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c2 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  height: 100%;
  width: 100%;
  opacity: 0;
  border: none;
  cursor: pointer;
}

.c2::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c2::-moz-focus-inner {
  border: none;
  outline: none;
}

.c4 {
  cursor: pointer;
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border: dashed 4px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

<div
  class="c0"
  tabindex="0"
>
  <div
    class="c1"
  >
    <input
      accept="image/*"
      class="c2"
      name="file"
      tabindex="-1"
      type="file"
    />
  </div>
  <div
    class="c3"
  >
    <div
      class="c4 c5"
    >
      <span
        class="c6"
      >
        Drop file here or
         
        <a
          class="c7"
        >
          browse
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`FileInput basic 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: dashed 4px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.c3 {
  position: relative;
  display: block;
  pointer-events: none;
}

.c6 {
  margin: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c2 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  height: 100%;
  width: 100%;
  opacity: 0;
  border: none;
  cursor: pointer;
}

.c2::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c2::-moz-focus-inner {
  border: none;
  outline: none;
}

.c4 {
  cursor: pointer;
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border: dashed 4px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

<div
  class="c0"
  tabindex="0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      name="file"
      tabindex="-1"
      type="file"
    />
  </div>
  <div
    class="c3"
  >
    <div
      class="c4 c5"
    >
      <span
        class="c6"
      >
        Drop file here or
         
        <a
          class="c7"
        >
          browse
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`FileInput disabled 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: dashed 4px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.c3 {
  position: relative;
  display: block;
  pointer-events: none;
}

.c6 {
  margin: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c2 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  height: 100%;
  width: 100%;
  opacity: 0;
  border: none;
  cursor: pointer;
}

.c2::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c2::-moz-focus-inner {
  border: none;
  outline: none;
}

.c4 {
  cursor: pointer;
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border: dashed 4px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

<div
  class="c0"
  tabindex="0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      disabled=""
      name="file"
      tabindex="-1"
      type="file"
    />
  </div>
  <div
    class="c3"
  >
    <div
      class="c4 c5"
    >
      <span
        class="c6"
      >
        Drop file here or
         
        <a
          class="c7"
        >
          browse
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`FileInput multiple 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: dashed 4px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.c3 {
  position: relative;
  display: block;
  pointer-events: none;
}

.c6 {
  margin: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c2 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  height: 100%;
  width: 100%;
  opacity: 0;
  border: none;
  cursor: pointer;
}

.c2::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c2::-moz-focus-inner {
  border: none;
  outline: none;
}

.c4 {
  cursor: pointer;
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border: dashed 4px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

<div
  class="c0"
  tabindex="0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      multiple=""
      name="file"
      tabindex="-1"
      type="file"
    />
  </div>
  <div
    class="c3"
  >
    <div
      class="c4 c5"
    >
      <span
        class="c6"
      >
        Drop files here or
         
        <a
          class="c7"
        >
          browse
        </a>
      </span>
    </div>
  </div>
</div>
`;
